src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -185,121 +166,187 @@ def __enter__(self):
+            bulk_logger.warning(f"The batch run timed out, with {len(self._result_dict)} line results processed.")
+        return [self._result_dict[key] for key in sorted(self._result_dict)]
 
-    def _handle_output_queue_messages(self, output_queue: Queue, result_list: List[LineResult]):
-        try:
-            message = output_queue.get(timeout=1)
-            if isinstance(message, LineResult):
-                message = self._process_multimedia(message)
-                result_list.append(message)
-                return message
-            elif isinstance(message, FlowRunInfo):
-                self._storage.persist_flow_run(message)
-                return message
-            elif isinstance(message, NodeRunInfo):
-                self._storage.persist_node_run(message)
-                return message
-        except queue.Empty:
-            pass
-        return None
+    # region monitor thread target function
 
     def _monitor_workers_and_process_tasks_in_thread(
         self,
         task_queue: Queue,
-        result_list: List[LineResult],
+        result_dict: Dict[int, LineResult],
         index: int,
         input_queue: Queue,
         output_queue: Queue,
-        batch_start_time: datetime,
     ):
-        index, process_id, process_name = self._get_process_info(index)
-
-        # Entering the while loop requires two conditions:
-        # 1. The task queue is not empty, meaning there are lines yet to be executed.
-        # 2. The batch run has not reached the batch timeout limit.
-        while not self._batch_timeout_expired(batch_start_time):
-            self._processes_manager.ensure_healthy()
-            try:
-                # Get task from task_queue
-                inputs, line_number, run_id = task_queue.get(timeout=1)
-            except queue.Empty:
-                break
+        # Get the process info of the thread monitoring from the manager.
+        index, process_id, process_name = self._processes_manager.get_process_info(index)
+        self._processes_manager.ensure_healthy()
 
-            # Calculate the line timeout for the current line.
-            line_timeout_sec = self._line_timeout_sec
-            if self._batch_timeout_sec:
-                remaining_execution_time = (
-                    self._batch_timeout_sec - (datetime.utcnow() - batch_start_time).total_seconds()
-                )
-                if remaining_execution_time <= 0:
-                    self._is_timeout = True
+        # The main loop of the thread, responsible for getting tasks from the task queue and
+        # processing them through the input queue, while also monitoring for terminate signals.
+        # Currently, it exits this loop only upon receiving a terminate signal (TERMINATE_SIGNAL).
+        exit_loop = False
+        while not exit_loop:
+            while True:
+                try:
+                    # Get task from task_queue
+                    data = task_queue.get(timeout=1)
```

Comment:
It looks like a sync wait, not an async wait?

Reply:
The `_monitor_workers_and_process_tasks_in_thread` is a sync function.